apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-editor-save-click.js

```diff
@@ -0,0 +1,16 @@
+import tracksRecordEvent from './track-record-event';
+
+/**
+ * Return the event definition object to track `wpcom_block_editor_save_click`.
+ *
+ * @returns {import('./types').DelegateEventHandler} event object definition.
+ */
+export default () => ( {
+	id: 'wpcom-block-editor-save-click',
+	// The first selector is for site editing. The second is for post editing.
+	selector: '.editor-entities-saved-states__save-button, .editor-post-publish-button',
```

Comment:
```suggestion
	selector: '.editor-entities-saved-states__save-button, .editor-post-publish-button:not(.has-changes-dot), .editor-post-save-draft', ,
```

It looks like there are a few edge cases in the post editor where we need add more selectors or conditions.

1. We can trigger the multi-entity-save panel in the post editor by editing a site block, template part, reusable block, etc. When this happens the publish button keeps the same selector but adds a `.has-changes-dot` class to denote it will open the multi entity save panel. Thus clicking on the publish button fires the event once while opening the save panel, then when the actual save button is pressed in this panel it fires the event again due to the entities-saved-states selector being present there. So adding `:not(.has-changes-dot)` SHOULD resolve this issue.
![Screen Shot 2022-01-25 at 2 10 23 PM](https://user-images.githubusercontent.com/28742426/151044989-54c6cf95-2cb8-4f57-b0b9-cfaaa2d5865b.png)

2. Before a page or post is published, we have both the 'save draft' and 'publish' buttons present. We should add the `.editor-post-save-draft` selector to cover the save draft / unpublished case.  The publish flow for saving in this state works as expected, the publish button that opens the panel is not firing the event, and the publish button in the panel that saves fires the event as expected.
![Screen Shot 2022-01-25 at 2 13 26 PM](https://user-images.githubusercontent.com/28742426/151045244-3ca77d02-13cb-4d25-a0c0-05643d5b3690.png)

We are relying on a lot more selectors now, so this event seems a bit less stable than it used to when we were only concerned with that one button in the site editor. However, these selectors in the post editor have not changed in a long time and I would not expect them to any time soon.

Reply:
Great catch. What you suggested makes sense to me. I'm testing the updates now 👍